include/deal.II/distributed/tria_base.h

```diff
@@ -76,6 +76,7 @@ namespace parallel
    * classes derived from the current one it actually is).
    */
   template <int dim, int spacedim = dim>
+  DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
   class TriangulationBase : public dealii::Triangulation<dim, spacedim>
```

Comment:
You will have to copy this annotation to all member function implementations as well.

Reply:
Oh wait, it's already there. That's wild.